src/rtcModule/webrtc.cpp

```diff
@@ -336,64 +370,69 @@ void RtcModule::removeCall(Call& call)
+                errors.append("Configured video input device '").append(mVideoInDeviceName)
+                      .append("' not present, using default device\n");
+            }
+
+            updateConstraints(resolution);
+            auto opts = std::make_shared<artc::MediaGetOptions>(*device, mMediaConstraints);
+            videoInput = mDeviceManager.getUserVideo(opts);
+        }
+        catch(exception& e)
         {
-            device = &devices.video[0];
-            errors.append("Configured video input device '").append(mVideoInDeviceName)
-                  .append("' not present, using default device\n");
+            videoInput.reset();
+            errors.append("Error getting video device: ")
+                  .append(e.what()?e.what():"Unknown error")+='\n';
         }
-        auto opts = std::make_shared<artc::MediaGetOptions>(*device, mMediaConstraints);
+     }
 
-        mVideoInput = mDeviceManager.getUserVideo(opts);
-    }
-    catch(exception& e)
-    {
-        mVideoInput.reset();
-        errors.append("Error getting video device: ")
-              .append(e.what()?e.what():"Unknown error")+='\n';
-    }
+    if (!devices.audio.empty() && !mAudioInDeviceName.empty())
+     {
+        try
+         {
+            auto device = getDevice(mAudioInDeviceName, devices.audio);
+            if (!device)
+            {
+                errors.append("Configured audio input device '").append(mAudioInDeviceName)
+                      .append("' not present, using default device\n");
+                device = &devices.audio[0];
+            }
 
-    if (devices.audio.empty() || mAudioInDeviceName.empty())
-    {
-        mAudioInput.reset();
-    }
-    else if (!mAudioInput || mAudioInput.mediaOptions().device.name != mAudioInDeviceName)
-    try
-    {
-        auto device = getDevice(mAudioInDeviceName, devices.audio);
-        if (!device)
-        {
-            errors.append("Configured audio input device '").append(mAudioInDeviceName)
-                  .append("' not present, using default device\n");
-            device = &devices.audio[0];
+            audioInput = mDeviceManager.getUserAudio(
+                    std::make_shared<artc::MediaGetOptions>(*device, mMediaConstraints));
         }
-        mAudioInput = mDeviceManager.getUserAudio(
-                std::make_shared<artc::MediaGetOptions>(*device, mMediaConstraints));
-    }
-    catch(exception& e)
-    {
-        mAudioInput.reset();
-        errors.append("Error getting audio device: ")
-              .append(e.what()?e.what():"Unknown error")+='\n';
-    }
-    if (!mAudioInput && !mVideoInput)
-        return std::make_shared<artc::LocalStreamHandle>(nullptr, nullptr);
+        catch(exception& e)
+        {
+            audioInput.reset();
+            errors.append("Error getting audio device: ")
+                  .append(e.what()?e.what():"Unknown error")+='\n';
+         }
+     }
+
+   if (!audioInput && !videoInput)
```

Comment:
No need for this sanity checkup. `setAv()` is already protected against null audio/video.